client/components/seo/preview-upgrade-nudge/test/index.jsx

```diff
@@ -32,57 +35,61 @@ const props = {
-		expect( comp.find( 'TrackComponentView' ).length ).toBe( 1 );
-		expect( comp.find( 'TrackComponentView' ).props().eventName ).toBe(
-			'calypso_seo_preview_upgrade_nudge_impression'
-		);
+		render( <SeoPreviewNudge { ...props } /> );
+		const track = screen.queryByTestId( 'track-component-view' );
+		expect( track ).toBeVisible();
+		expect( JSON.parse( track.textContent ) ).toMatchObject( {
+			eventName: 'calypso_seo_preview_upgrade_nudge_impression',
+		} );
 	} );
 } );
 
 describe( 'UpsellNudge should get appropriate plan constant', () => {
-	[ PLAN_FREE, PLAN_BLOGGER, PLAN_PERSONAL, PLAN_PREMIUM ].forEach( ( product_slug ) => {
-		test( `Pro 1 year for (${ product_slug })`, () => {
-			const comp = shallow(
+	test.each( [ PLAN_FREE, PLAN_BLOGGER, PLAN_PERSONAL, PLAN_PREMIUM ] )(
+		`Pro 1 year for (%s)`,
+		( product_slug ) => {
+			render(
 				<SeoPreviewNudge { ...props } isJetpack={ false } site={ { plan: { product_slug } } } />
 			);
-			expect( comp.find( 'UpsellNudge' ).length ).toBe( 1 );
-			expect( comp.find( 'UpsellNudge' ).props().plan ).toBe( PLAN_BUSINESS );
-		} );
-	} );
+			const nudge = screen.queryByTestId( 'upsell-nudge' );
+			expect( nudge ).toBeVisible();
+			expect( JSON.parse( nudge.textContent ) ).toMatchObject( { plan: PLAN_BUSINESS } );
+		}
+	);
 
-	[ PLAN_BLOGGER_2_YEARS, PLAN_PERSONAL_2_YEARS, PLAN_PREMIUM_2_YEARS ].forEach(
+	test.each( [ PLAN_BLOGGER_2_YEARS, PLAN_PERSONAL_2_YEARS, PLAN_PREMIUM_2_YEARS ] )(
+		`Business 2 year for (%s)`,
 		( product_slug ) => {
-			test( `Business 2 year for (${ product_slug })`, () => {
-				const comp = shallow(
-					<SeoPreviewNudge { ...props } isJetpack={ false } site={ { plan: { product_slug } } } />
-				);
-				expect( comp.find( 'UpsellNudge' ).length ).toBe( 1 );
-				expect( comp.find( 'UpsellNudge' ).props().plan ).toBe( PLAN_BUSINESS_2_YEARS );
-			} );
+			render(
+				<SeoPreviewNudge { ...props } isJetpack={ false } site={ { plan: { product_slug } } } />
+			);
+			const nudge = screen.queryByTestId( 'upsell-nudge' );
+			expect( nudge ).toBeVisible();
+			expect( JSON.parse( nudge.textContent ) ).toMatchObject( { plan: PLAN_BUSINESS_2_YEARS } );
 		}
 	);
 
-	[
+	test.each( [
 		PLAN_JETPACK_FREE,
 		PLAN_JETPACK_PERSONAL,
 		PLAN_JETPACK_PERSONAL_MONTHLY,
 		PLAN_JETPACK_PREMIUM,
 		PLAN_JETPACK_PREMIUM_MONTHLY,
 		PLAN_JETPACK_BUSINESS_MONTHLY,
 		PLAN_JETPACK_SECURITY_DAILY_MONTHLY,
-	].forEach( ( product_slug ) => {
-		test( `Jetpack Business for (${ product_slug })`, () => {
-			const comp = shallow(
-				<SeoPreviewNudge { ...props } isJetpack={ true } site={ { plan: { product_slug } } } />
-			);
-			expect( comp.find( 'UpsellNudge' ).length ).toBe( 1 );
-			expect( comp.find( 'UpsellNudge' ).props().plan ).toBe( PLAN_JETPACK_SECURITY_DAILY );
+	] )( `Jetpack Business for (%s)`, ( product_slug ) => {
+		render(
+			<SeoPreviewNudge { ...props } isJetpack={ true } site={ { plan: { product_slug } } } />
```

Comment:
There might be a few of those redundant `={ true }` in those tests - a good opportunity to remove them.
```suggestion
			<SeoPreviewNudge { ...props } isJetpack site={ { plan: { product_slug } } } />
```